### PR TITLE
Add routes resource & attributes

### PR DESCRIPTION
**Pull request by rustamwin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️/❌
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

File: src/Resource/ResourceInterface.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Yiisoft\Router\Resource;
+
+use Yiisoft\Router\Group;
+use Yiisoft\Router\Route;
+
+/**
+ * `ResourceInterface` is a resource of routes.
+ */
+interface ResourceInterface
```
Comment (vjik) [inline src/Resource/ResourceInterface.php:13]:
```suggestion
interface RoutesProviderInterface
```

File: src/Resource/AttributeResource.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+namespace Yiisoft\Router\Resource;
+
+use olvlvl\ComposerAttributeCollector\Attributes;
+use Yiisoft\Router\Group;
+use Yiisoft\Router\Route;
+
+/**
+ * An attribute resource represents routes that declared via PHP Attributes.
+ * @codeCoverageIgnore
```
Comment (vjik) [inline src/Resource/AttributeResource.php:13]:
```suggestion
 * @codeCoverageIgnore
```

Why ignore?

File: src/RouteCollector.php [hunk 3]
```diff
@@ -25,6 +32,12 @@ public function addRoute(Route|Group ...$routes): RouteCollectorInterface
         return $this;
     }
 
+    public function addResource(ResourceInterface $resource): RouteCollectorInterface
```
Comment (vjik) [inline src/RouteCollector.php:35]: It need to add resources in runtime? If no, better pass resources array to constructor.

File: src/Resource/AttributeResource.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+     */
+    private static array $reflectionsCache = [];
+
+    public function getRoutes(): array
+    {
+        $routes = [];
+        $groupRoutes = [];
+        $routePredicate = Attributes::predicateForAttributeInstanceOf(Route::class);
+        $targetMethods = Attributes::filterTargetMethods($routePredicate);
+        foreach ($targetMethods as $targetMethod) {
```
Comment (vjik) [inline src/Resource/AttributeResource.php:28]: Is it parse all PHP files? Can we search in a specific folder?

File: src/RouteCollector.php (hunk 3 above)
Comment (rustamwin) [reply src/RouteCollector.php]: Yes, it will be added in runtime like routes.

File: src/Resource/AttributeResource.php (hunk 4 above)
Comment (rustamwin) [reply src/Resource/AttributeResource.php]: Yes, it's a composer plugin, can be configured in composer.json. We may change this implementation in the future (perhaps when `yiisoft/classifier` is ready).

**Pull request merged**